ScreenManager.c

```diff
@@ -397,6 +397,22 @@ void ScreenManager_run(ScreenManager* this, Panel** lastFocus, int* lastKey, con
       case KEY_F(10):
          quit = true;
          continue;
+      case KEY_F(4):
+         redraw = false;
+         char msg[1024];
+         int saveOk = Settings_write(this->settings, false);
+         if (saveOk < 0) {
+            xSnprintf(msg, 1024, "Error saving to %s: %s", this->settings->filename, strerror(-saveOk));
+            attrset(CRT_colors[FAILED_SEARCH]);
+         } else {
+            xSnprintf(msg, 1024, "Saved to %s", this->settings->filename);
+            attrset(CRT_colors[FUNCTION_BAR]);
+         }
+
+         mvhline(LINES - 1, 0, ' ', COLS);
+         mvaddstr(LINES - 1, 0, msg);
+         attrset(CRT_colors[RESET_COLOR]);
```

Comment:
@fasterit @natoscott Not sure about this part. Technically correct, but …

* The wording should probably be more clear? `Successfully saved configuration to %s` vs. `Failed to save configuration to %s`.
* (Mis)using the function bar in this way seems kinda dirty.
* Error messages should be mirrored to `stderr`, so they are included in crash reports (just in case).

Reply:
This is solving a very, very niche problem. Hence I don't like this PR.
@arp242: Please just `chmod -w` your `htoprc` file.
/DLange

Reply:
But then I can never save/update it without mucking about with file permissions, and provisioning new machines is more difficult.

I don't know how "niche" it is; I'm pretty sure I added it to me ~/TODO a few years back after I saw some other people complain about it too, I think it was on Lobsters, but I can't find it right now; did find some people complaining about it on HN: https://news.ycombinator.com/item?id=24344542

The automatic saving is pretty unconventional; I don't know of any other (terminal) program that works like this. And when I first used htop I was pretty confused by it ("did it save? How does this work?") Even without a setting, an explicit "Save + feedback" is a good thing to have IMO.

> (Mis)using the function bar in this way seems kinda dirty.

Yeah, but I don't know where else it can be printed. The header can be hidden now, so you can't really use that, and this is the only thing available really.

Reply:
> Please just `chmod -w` your `htoprc` file.

On top ot the above concerns, htop will actually complain if you do this:
```
Can not save configuration to /home/novenary/.config/htop/htoprc: Permission denied
```

Reply:
> > Please just `chmod -w` your `htoprc` file.
> 
> On top ot the above concerns, htop will actually complain if you do this:
> 
> ```
> Can not save configuration to /home/novenary/.config/htop/htoprc: Permission denied
> ```

Yes, of course, because the expected default for nearly two decades now is that your configuration is saved.